src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -35,90 +18,42 @@
 
 public class ChannelSinkTask extends SinkTask {
     private static final Logger logger = LoggerFactory.getLogger(ChannelSinkTask.class);
-    private static final String[] severities = new String[]{"", "", "VERBOSE", "DEBUG", "INFO", "WARN", "ERROR", "ASSERT"};
 
-    private AblyRealtime ably;
+    private AblyClientFactory ablyClientFactory = new DefaultAblyClientFactory();
+    private AblyClient ablyClient;
 
-    private ChannelSinkMapping channelSinkMapping;
-    private MessageSinkMapping messageSinkMapping;
+    public ChannelSinkTask() {}
+    //for unit tests
+    public ChannelSinkTask(AblyClientFactory factory) {
+        this.ablyClientFactory = factory;
+    }
```

Comment:
You've added the default constructor and the one that accepts a `factory`. The second one is meant to be used for unit tests, but I can't find any unit tests for this class in this PR :thinking: Are you going to test this class and add the tests in some subsequent PRs? If not then this looks like a YAGNI :wink: However, if you will add some tests then maybe let's add a proper javadoc comments to both constructors that will explain their purpose and usage in the project? :thinking:

Reply:
Also, if this constructor is for unit testing then I would expect conventionally that you would remove the `public` accessibility modifier to make it (implicit) package-scoped, given the unit test should also be in this same package.

Reply:
@KacperKluka Yes I was intending to add tests later when I had a bit more time. Plus planning for the future, I was embracing that this could be used for custom implementation and mapping scenarios where clients could provide their own clients and mappings etc. 
@QuintinWillison Good point. I have made the change here https://github.com/ably/kafka-connect-ably/pull/63/commits/c0bd2a328ecf8282259c696beac48b5ad7d6e768. Also added a visibility annotation to make it a bit more clear it's for testing